src/server/internalServer.cpp

```diff
@@ -513,8 +513,11 @@ std::unique_ptr<Response> InternalServer::handle_search(const RequestContext& re
   } catch (const std::out_of_range&) {}
 
   /* Make the search */
-  if ( (!archive && !bookName.empty())
-    || (patternString.empty() && ! has_geo_query) ) {
+  if (patternString.empty() && ! has_geo_query) {
+    return Response::build_400(*this, "No query provided.");
```

Comment:
Here `"No query provided."` serves as the `url` parameter

Reply:
I've split the change in several commit and miss this.

This is fixed, directly in the new commit.